packages/apollo/src/utils/client.ts

```diff
@@ -0,0 +1,324 @@
+import { ApolloClient, from } from '@apollo/client'
+import { createUploadLink } from 'apollo-upload-client'
+import isEqual from 'lodash/isEqual'
+import { useMemo, useState } from 'react'
+
+import { isSSR } from '@open-condo/miniapp-utils/helpers/environment'
+import { useEffectOnce } from '@open-condo/miniapp-utils/hooks/useEffectOnce'
+import { usePrevious } from '@open-condo/miniapp-utils/hooks/usePrevious'
+
+
+import { addCacheToClient, createApolloCache } from './cache'
+import { createPersistor } from './cachePersistor'
+
+import { APOLLO_STATE_PROP_NAME } from '../constants'
+
+import type { InitCacheConfig, InitCacheOptions, InvalidationCacheConfig } from './cache'
+import type { CachePersistor } from './cachePersistor'
+import type { ApolloLink, NormalizedCacheObject } from '@apollo/client'
+
+type ApolloClientOptions = {
+    /** Apollo server url, can be static string or obtained from function on the fly */
+    uri: (() => string) | string
+    /** Headers to include in each request, can be helpful in SSR environments */
+    headers?: Record<string, string>
+    /** Client's cache config */
+    cacheConfig?: InvalidationCacheConfig
+    /** Middleware to process / enrich requests */
+    middlewares?: Array<ApolloLink>
+}
+
+export type InitApolloClientOptions = Partial<Omit<ApolloClientOptions, 'cacheConfig'>> & {
+    /** Results in cache being initialized with skipOnRead */
+    networkOnly?: boolean
+}
+
+export type ApolloHelperOptions = Omit<ApolloClientOptions, 'cacheConfig'> & {
+    /** Cache initializer */
+    cacheConfig?: InitCacheConfig
+}
+
+type SSRResult<PropsType> = {
+    props: PropsType & {
+        [APOLLO_STATE_PROP_NAME]?: NormalizedCacheObject
+    }
+}
+
+type UseApolloHookResult = {
+    client: ApolloClient<NormalizedCacheObject>
+    cachePersistor: CachePersistor<NormalizedCacheObject> | undefined
+}
+
+/**
+ * Creates new apollo client with InvalidationCache
+ */
+function createApolloClient (options: ApolloClientOptions): ApolloClient<NormalizedCacheObject> {
+    const middlewares = options.middlewares || []
+    const httpLink = createUploadLink({
+        uri: typeof options.uri === 'string' ? options.uri : options.uri(),
+        headers: options.headers,
+        credentials: 'include',
+        fetchOptions: {
+            mode: 'cors',
+        },
+    }) as unknown as ApolloLink
```

Comment:
Can I override other options for `httpLink` or use my own custom one?

Reply:
Which one do you want to override?

Reply:
I don't want to override anything)
But why don't we give such an opportunity to other?

Reply:
Here is a complete list of options that can be passed: https://github.com/jaydenseric/apollo-upload-client/blob/5e6d5c17861fbd9e2d84c8517fc29d5321be4cfc/createUploadLink.mjs#L80

Reply:
I guess it's a good practice to start with a bare minimum amount of parameters and extend it on request when needed. I'd suggest to limit it right now, since we don't need anything else at this time and figure out its growth later